### PR TITLE
Ported to Qt6 and used layer-shell-qt

### PR DESCRIPTION
**Pull request by tsujan**
Porting to Qt6 and using layer-shell-qt are dome simultaneously to make lxqt-runner completely ready for Wayland.

### CODE REVIEW

**Comment by stefonarch** [comment]
Working fine - removing the last window rules from my wayland configurations now.

**Comment by tsujan** [comment]
If you find the time, please also test it under X11 (I will later). After all, 99% percent of LXQt users have X11 sessions ;)

File: dialog.cpp [hunk 1]
```diff
@@ -201,9 +204,43 @@ void Dialog::moveEvent(QMoveEvent *event)
...
-    return QDialog::showEvent(event);
+    if (QGuiApplication::platformName() == QSL("wayland"))
+    {
+        winId();
+        if (QWindow *win = windowHandle())
+        {
+            if (LayerShellQt::Window* layershell = LayerShellQt::Window::get(win))
+            {
+                layershell->setLayer(LayerShellQt::Window::Layer::LayerTop);
+                layershell->setKeyboardInteractivity(LayerShellQt::Window::KeyboardInteractivityOnDemand);
```
Comment (stefonarch) [inline dialog.cpp:215]: Most probably we need to set this to `KeyboardInteractivityExclusive` as in sway and latest labwc it only gets focus by mouse action.

**Review by stefonarch — CHANGES_REQUESTED** [review]
See https://github.com/labwc/labwc/issues/1653#issuecomment-2016743861

File: dialog.cpp (hunk 1 above)
Comment (tsujan) [reply dialog.cpp:215]: The problem should be fixed in labwc (although its latest stable version has no problem), and "on-demand" should work: the window is shown and demands focus; there's noting "exclusive" about it.

Comment (stefonarch) [reply dialog.cpp:215]: To me "exclusive" seems more logic, but if we stick with `OnDemand` we need to open a bug report in sway (and maybe others).

Comment (tsujan) [reply dialog.cpp:215]: If "exclusive" isn't just a word (it seems to be until now), I don't think we should use it here. Runner doesn't get the focus exclusively; it just get the focus when shown.

Comment (tsujan) [reply dialog.cpp:215]: See https://wayland.app/protocols/wlr-layer-shell-unstable-v1#zwlr_layer_surface_v1:enum:keyboard_interactivity:entry:exclusive. I don't like this part:

"In this case, no guarantee is made when this surface will receive keyboard focus (if ever)."

In contrast, read the explanation of `on_demand` below it. It sounds quite reasonable for us.

Comment (stefonarch) [reply dialog.cpp:215]: That sounds convincing to leave it as it is. Opening a bug at sway when this is merged.

**Comment by stefonarch** [comment]
No idea how to remove that "changes requested though.

**Review by stefonarch — APPROVED** [review]

**Comment by tsujan** [comment]
> See labwc/labwc#1653 (comment)

Yes, I'd seen it. It worried me, to be honest ;) You know that I love labwc (I'm using it right now), but changing something in our code just to make it work under its git source doesn't seem OK to me.

**Comment by tsujan** [comment]
> No idea how to remove that "changes requested though.

If you want to remove it without approving, you could press the "Re-quest review" beside your avatar on a PR page.

**Comment by Chiitoo** [comment]
Quick test only, but seems good so far on X11 here (with https://github.com/lxqt/lxqt-globalkeys/pull/278).

Thank you!

**Comment by Consolatis** [comment]
> > See [labwc/labwc#1653](https://github.com/labwc/labwc/issues/1653) (comment)
> 
> Yes, I'd seen it. It worried me, to be honest [..] changing something in our code just to make it work under its git source doesn't seem OK to me.

That is a very good stance from my point of view as well.
We should try to ensure that all sides respect the protocol as close as possible and not write applications that work for compositor `x` and `y` but requires changes for `z` but rather implement the protocol and raise issues where a compositor (or application) has the wrong assumptions or behavior.

**Pull request merged**